extensions/ql-vscode/src/view/model-editor/MethodRow.tsx

```diff
@@ -55,10 +63,10 @@ const DataGridRow = styled(VSCodeDataGridRow)<{ focused?: boolean }>`
 export type MethodRowProps = {
   method: Method;
   methodCanBeModeled: boolean;
-  modeledMethod: ModeledMethod | undefined;
+  modeledMethods: ModeledMethod[];
   methodIsUnsaved: boolean;
   modelingInProgress: boolean;
-  mode: Mode;
+  viewState: ModelEditorViewState;
   revealedMethodSignature: string | null;
   onChange: (modeledMethod: ModeledMethod) => void;
```

Comment:
I'm not yet sure what the signature for `onChange` should be. Perhaps it should not be `(modeledMethods: ModeledMethod[]) => void` so we send over all the modelings for this method whenever anything changes.

I think it'll have to be that, as otherwise is something changes on one modeling, the receiver of the callback wouldn't know which modeling it was. So we have to send all of them at once so it knows if the number of modelings has changed and the exact state of all of them.

I could do that in this PR or the next one. This PR isn't intending to actually make this component work yet and be able to edit multiple modelings. I just want to get the UI looking right and then we can hook up all the implementation later. So long as it still works correctly when the feature flag is disabled that's ok.

Reply:
I think sending all of them even if only 1 has changed makes the most sense. My suggestion would be something like `onChange: (signature: string, modeledMethods: ModeledMethod[])`. I would suggest adding the signature so we don't need to find the signature of the method that is being modeled from the `modeledMethods`, but can simply use it as-is.